src/v/kafka/server/handlers/offset_for_leader_epoch.cc

```diff
@@ -80,7 +80,7 @@ static ss::future<std::vector<epoch_end_offset>> fetch_offsets(
         ret.push_back(response_t::make_epoch_end_offset(
           r.ktp.get_partition(),
           co_await get_epoch_end_offset(r.requested_epoch, *p),
-          p->leader_epoch()));
+          r.requested_epoch));
```

Comment:
Nice! Looks like this has been around for a while and should be backported https://github.com/redpanda-data/redpanda/blob/v22.2.x/src/v/kafka/server/handlers/offset_for_leader_epoch.cc#L89